test/lit/passes/code-pushing_into_if.wast

```diff
@@ -0,0 +1,540 @@
+    )
+  )
+
+  ;; CHECK:      (func $if-use (param $p i32)
+  ;; CHECK-NEXT:  (local $x i32)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.get $p)
+  ;; CHECK-NEXT:   (block
+  ;; CHECK-NEXT:    (local.set $x
+  ;; CHECK-NEXT:     (i32.const 1)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $if-use (param $p i32)
+    (local $x i32)
+    ;; The set local is used in one arm and nowhere else; push it there.
+    (local.set $x (i32.const 1))
+    (if
+      (local.get $p)
+      (drop (local.get $x))
+    )
+  )
+
+  ;; CHECK:      (func $if-use-nop (param $p i32)
+  ;; CHECK-NEXT:  (local $x i32)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.get $p)
+  ;; CHECK-NEXT:   (block
+  ;; CHECK-NEXT:    (local.set $x
+  ;; CHECK-NEXT:     (i32.const 1)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (nop)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $if-use-nop (param $p i32)
+    (local $x i32)
+    (local.set $x (i32.const 1))
+    (if
+      (local.get $p)
+      (drop (local.get $x))
+      (nop) ;; add a nop here compared to the last testcase (no output change)
+    )
+  )
+
+  ;; CHECK:      (func $if-else-use (param $p i32)
+  ;; CHECK-NEXT:  (local $x i32)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.get $p)
+  ;; CHECK-NEXT:   (nop)
+  ;; CHECK-NEXT:   (block
+  ;; CHECK-NEXT:    (local.set $x
+  ;; CHECK-NEXT:     (i32.const 1)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $if-else-use (param $p i32)
+    (local $x i32)
+    ;; The set local is used in one arm and nowhere else; push it there.
+    (local.set $x (i32.const 1))
+    (if
+      (local.get $p)
+      (nop)
+      (drop (local.get $x))
+    )
+  )
+
+  ;; CHECK:      (func $unpushed-interference (param $p i32)
+  ;; CHECK-NEXT:  (local $x i32)
+  ;; CHECK-NEXT:  (local $y i32)
+  ;; CHECK-NEXT:  (local.set $x
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (local.set $y
+  ;; CHECK-NEXT:   (local.get $x)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.get $p)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (local.get $x)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $unpushed-interference (param $p i32)
+    (local $x i32)
+    (local $y i32)
+    (local.set $x (i32.const 1))
+    ;; This set is not pushed (as it is not used in the if) and it will then
+    ;; prevent the previous set of $x from being pushed, since we can't push a
+    ;; set of $x past a get of it.
+    (local.set $y (local.get $x))
```

Comment:
How about a test case where $y is pushed into one arm and $x would be pushed into the other arm except for the get of $x that went along with $y?

Reply:
Good catch! Looks like that's a bug in the current logic.

Reply:
Fixed.